src/PuJetIdSF.cc

```diff
@@ -1,31 +1,39 @@
 #include "PuJetIdSF.h"
 
+#define DEBUG false
+
 // Constructor
 PuJetIdSF::PuJetIdSF (std::string directory, std::string year)
 {
   // Build file names
-  std::string nf_eff    = directory + "/h2_eff_mc_" + year + "_L.root";
-  std::string nf_eff_sf = directory + "/h2_eff_sf_" + year + "_L.root";
+  std::string nf_eff       = directory + "/h2_eff_mc_" + year + "_L.root";
+  std::string nf_eff_sf    = directory + "/h2_eff_sf_" + year + "_L.root";
   std::string nf_mistag    = directory + "/h2_mistag_mc_" + year + "_L.root";
   std::string nf_mistag_sf = directory + "/h2_mistag_sf_" + year + "_L.root";
+  std::string nf_sf_err    = directory + "/scalefactorsPUID_81Xtraining.root";
 
   // Open files
-  TFile* f_eff    = new TFile(nf_eff.c_str(),"READ");
-  TFile* f_eff_sf = new TFile(nf_eff_sf.c_str(),"READ");
+  TFile* f_eff       = new TFile(nf_eff.c_str(),"READ");
+  TFile* f_eff_sf    = new TFile(nf_eff_sf.c_str(),"READ");
   TFile* f_mistag    = new TFile(nf_mistag.c_str(),"READ");
   TFile* f_mistag_sf = new TFile(nf_mistag_sf.c_str(),"READ");
+  TFile* f_sf_err    = new TFile(nf_sf_err.c_str(),"READ");
 
   // Build TH2F names
-  std::string nh_eff    = "h2_eff_mc"+year+"_L";
-  std::string nh_eff_sf = "h2_eff_sf"+year+"_L";
-  std::string nh_mistag    = "h2_mistag_mc"+year+"_L";
-  std::string nh_mistag_sf = "h2_mistag_sf"+year+"_L";
+  std::string nh_eff           = "h2_eff_mc"+year+"_L";
+  std::string nh_eff_sf        = "h2_eff_sf"+year+"_L";
+  std::string nh_eff_sf_err    = "h2_eff_sf"+year+"_L_Systuncty";
+  std::string nh_mistag        = "h2_mistag_mc"+year+"_L";
+  std::string nh_mistag_sf     = "h2_mistag_sf"+year+"_L";
+  std::string nh_mistag_sf_err = "h2_mistag_sf"+year+"_L_Systuncty";
 
   // Get the TH2F histograms
-  h_eff_    = (TH2F*) f_eff->Get(nh_eff.c_str());
-  h_eff_SF_ = (TH2F*) f_eff_sf->Get(nh_eff_sf.c_str());
-  h_mistag_    = (TH2F*) f_mistag->Get(nh_mistag.c_str());
-  h_mistag_SF_ = (TH2F*) f_mistag_sf->Get(nh_mistag_sf.c_str());
+  h_eff_           = (TH2F*) f_eff->Get(nh_eff.c_str());
+  h_eff_SF_        = (TH2F*) f_eff_sf->Get(nh_eff_sf.c_str());
+  h_eff_SF_err_    = (TH2F*) f_sf_err->Get(nh_eff_sf_err.c_str());
+  h_mistag_        = (TH2F*) f_mistag->Get(nh_mistag.c_str());
+  h_mistag_SF_     = (TH2F*) f_mistag_sf->Get(nh_mistag_sf.c_str());
+  h_mistag_SF_err_ = (TH2F*) f_sf_err->Get(nh_mistag_sf_err.c_str());
 }
 
 
```

Comment:
Add deletion of new histograms in the destructor

Reply:
thanks! I will add them!